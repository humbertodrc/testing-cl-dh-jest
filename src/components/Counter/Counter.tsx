import React from "react";
import {useState} from "react";

export interface CounterInterface {
	value: number;
}

const Counter: React.FC<CounterInterface> = ({value}) => {
	const [count, setCount] = useState(value);
	return (
		<div className="App">
			<h1>Testing React</h1>
			<h2 data-aria="number">{count}</h2>
			<div className="card">
				<button onClick={() => setCount((count) => count + 1)}>+1</button>
				<button onClick={() => setCount((count) => count - 1)}>-1</button>
			</div>
		</div>
	);
};

export default Counter;
