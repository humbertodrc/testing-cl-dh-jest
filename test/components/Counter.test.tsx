import {fireEvent, render, screen} from "@testing-library/react";
import React from "react";
import Counter from "../../src/components/Counter/Counter";

describe("Pruebas en <Counter />", () => {
	const initialValue = 10;

	test("Debe hacer match con el snapshot", () => {
		const {container} = render(<Counter value={initialValue} />);

		expect(container).toMatchSnapshot();
	});

	test("Debe Mostrar el Valor inicial de 50", () => {
		render(<Counter value={50} />);

		screen.debug();

		expect(screen.getByText("50")).toBeTruthy();

		expect(screen.getByRole("heading", {level: 2}).innerHTML).toContain("50");
	});

	test("Debe incrementar el valor +1", () => {
		render(<Counter value={initialValue} />);

		const btnIncrement = screen.getByText("+1");

		fireEvent.click(btnIncrement);

		expect(screen.getByText("11")).toBeTruthy();
	});

	test("Debe decrementar el valor -1", () => {
		render(<Counter value={initialValue} />);

		const btnDecrement = screen.getByText("-1");

		fireEvent.click(btnDecrement);

		expect(screen.getByText("9")).toBeTruthy();
	});
});
