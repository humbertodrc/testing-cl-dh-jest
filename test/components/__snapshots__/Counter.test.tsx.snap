// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Counter /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Testing React
    </h1>
    <h2
      data-aria="number"
    >
      10
    </h2>
    <div
      class="card"
    >
      <button>
        +1
      </button>
      <button>
        -1
      </button>
    </div>
  </div>
</div>
`;
